### PR TITLE
Data sachplan militaer

### PR DESCRIPTION
**Pull request by AFoletti**

### CODE REVIEW

File: chsdi/models/vector/vbs.py [hunk 1]
```diff
@@ -50,3 +50,201 @@ class TERRITORIALREGIONEN(Base, Vector):
...
+    validfrom = Column('validfrom', Text)
+    description_de = Column('description_de', Text)
+    description_fr = Column('description_fr', Text)
+    description_it = Column('description_it', Text)
+    doc_web = Column('doc_web', Text)
+    objname_de = Column('objname_de', Text)
+    objname_fr = Column('objname_fr', Text)
+    objname_it = Column('objname_it', Text)
+    bgdi_created = Column('bgdi_created', Text)
+    the_geom = GeometryColumn(Geometry(dimension=2, srid=21781))
```
Comment (loicgasser) [inline chsdi/models/vector/vbs.py]: Since today because of some slq/geoalchemy updates you'll need to rebase and adapt geometry column declaration.

**Comment by cedricmoullet** [comment]
@AFoletti is it ready ?

**Comment by AFoletti** [comment]
No. This integration is only a POC. We are solving several geometry problems and a demo to the datenherr is planned for March

File: chsdi/models/vector/vbs.py [hunk 2]
```diff
@@ -54,3 +54,209 @@ class TERRITORIALREGIONEN(Base, Vector):
     name = Column('name', Text)
 
 register('ch.vbs.territorialregionen', TERRITORIALREGIONEN)
+
+
+class sim_facilities_a(Base, Vector):
```
Comment (procrastinatio) [inline chsdi/models/vector/vbs.py]: From PEP8 _Class names should normally use the CapWords convention_

Comment (AFoletti) [inline chsdi/models/vector/vbs.py]: see https://github.com/geoadmin/mf-chsdi3/issues/1315

**Comment by cedricmoullet** [comment]
@AFoletti what do we do with this PR ?

**Comment by AFoletti** [comment]
we wait...
We should receive the actual data during the next couple of months, and the GoLive is planned somewhere near september

**Comment by alcapat** [comment]
jenkins test

**Comment by cedricmoullet** [comment]
@AFoletti ???

**Comment by AFoletti** [comment]
I could copy-paste my last answer ;)

Still a POC. The military was supposed to give us definitive data and a comprehensive feedback last september but the only thing they delivered is a "we don't know yet".

**Comment by procrastinatio** [comment]
Close that PR it is not good for karma and stats.

**Comment by ltclm** [comment]
the layer will be published 2016-08-31

File: chsdi/models/vector/vbs.py [hunk 3]
```diff
@@ -160,3 +160,203 @@ class Waldschaden(Base, Vector):
     y_koord = Column('y_koord', Integer)
 
 register('ch.vbs.waldschadenkarte', Waldschaden)
+
+
+class sim_facilities_a(Base, Vector):
```
Comment (loicgasser) [inline chsdi/models/vector/vbs.py]: Class name in camel case please

File: chsdi/models/vector/vbs.py [hunk 4]
```diff
@@ -160,3 +160,203 @@ class Waldschaden(Base, Vector):
...
+    facstatus_text_it = Column('facstatus_text_it', Text)
+    validfrom = Column('validfrom', Text)
+    description_de = Column('description_de', Text)
+    description_fr = Column('description_fr', Text)
+    description_it = Column('description_it', Text)
+    doc_web = Column('doc_web', Text)
+    objname_de = Column('objname_de', Text)
+    objname_fr = Column('objname_fr', Text)
+    objname_it = Column('objname_it', Text)
+    bgdi_created = Column('bgdi_created', Text)
```
Comment (loicgasser) [inline chsdi/models/vector/vbs.py]: We have a lot of duplicated attributes. Please create a base class and extend if needed.
https://github.com/geoadmin/mf-chsdi3/blob/master/chsdi/models/vector/uvek.py#L305
for inspiration.

**Comment by loicgasser** [comment]
2 comments otherwise LGTM

File: chsdi/models/vector/vbs.py (hunk 4 above)
Comment (loicgasser) [inline chsdi/models/vector/vbs.py]: Unicode instead of Text

Comment (ltclm) [inline chsdi/models/vector/vbs.py]: can we replace
```
from sqlalchemy import Column, Text, Integer
```
by
```
from sqlalchemy import Column, Unicode, Integer
```

File: chsdi/models/vector/vbs.py [hunk 5]
```diff
@@ -162,201 +162,106 @@ class Waldschaden(Base, Vector):
...
+    __tablename__ = 'sim_fac_anhoerung'
+    id = Column('stabil_id', Unicode, primary_key=True)
+
+register('ch.vbs.sachplan-infrastruktur-militaer_anhorung', SimFacilitiesA)
 
-class sim_facilities_k(Base, Vector):
+
+class SimFacilitiesK(Base, SimFacilities, Vector):
     __tablename__ = 'sim_fac_kraft'
+    id = Column('stabil_id', Integer, primary_key=True)
```
Comment (loicgasser) [inline chsdi/models/vector/vbs.py]: Ids of the same layer should be of the same type.

**Comment by loicgasser** [comment]
Ok 👌 squash rebase and merge

**Comment by ltclm** [comment]
will have to add custom legends

**Comment by loicgasser** [comment]
LGTM

**Pull request merged**